### DISCUSSION TITLE
How can I make Tridactyl focus on the page instead of address bar when a new tab is opened?

### DISCUSSION THREAD

**Discussion by realbaylen** [tridactyl/tridactyl#4967] [Q&A]
Thanks

**Comment by bovine3dom** [comment]
Open tabs directly with t / `:tabopen` rather than ctrl+t :)

w opens windows

**Reply by thetwistedlogic** [reply]
Hey. 

Sometimes, I need to use ctrl+t (for example on pages that I blacklist tridactyl). In that case, Is there anyway to focus on page instead of address bar? I'm not entirely sure if this is a tridactyl issue.

**Reply by bovine3dom** [reply]
once you've opened the page you can press F6 or ctrl+comma to get focus onto the page, but i don't know any way of doing it by default

**Reply by Chic-Tweetz** [reply]
I have an [AutoConfig](https://support.mozilla.org/en-US/kb/customizing-firefox-using-autoconfig) file for this. This is the script:

```
// First line must be a comment
try {
  let { BrowserWindowTracker } = ChromeUtils.importESModule("resource:///modules/BrowserWindowTracker.sys.mjs");
  let { PrivateBrowsingUtils } = ChromeUtils.importESModule("resource://gre/modules/PrivateBrowsingUtils.sys.mjs");

  // focus the page (instead of omnibar) when opening a new tab
  Services.obs.addObserver((event) => {
    BrowserWindowTracker.getTopWindow().gBrowser.selectedBrowser.focus();
  }, "browser-open-newtab-start");

  // and when opening a new window
  Services.obs.addObserver((event) => {
    let win = BrowserWindowTracker.getTopWindow();

    // new private windows can't have a custom homepage (or be hinted with tridactyl) so better to focus the omnibar
    if (!PrivateBrowsingUtils.isWindowPrivate(win)) {
        win.gBrowser.selectedBrowser.focus();
    }
  }, "browser-delayed-startup-finished");
} catch(e) { Components.utils.reportError(e); }
```

**Reply by thetwistedlogic** [reply]
@Chic-Tweetz wow. that works perfectly. thank you so much!

**Reply by philippdieter** [reply]
It took me a while to figure out, to make this work for me in Firefox version 137 I needed to disable the sandbox in autoconfig.js, so my file looks like this:
```
pref("general.config.filename", "firefox.cfg");
pref("general.config.obscure_value", 0);
pref("general.config.sandbox_enabled", false);
```

**Reply by philippdieter** [reply]
After using the autoconfig script for a while now  I noticed that it does not work when using https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/ and open a new tab in a container. I tried to understand why but it seems that I don't know about the internals of Firefox. For some reason the observer `browser-open-newtab-start` is not triggered.

Does anybody have an idea how to solve this?

**Reply by sk9hi** [reply]
@Chic-Tweetz ty for the script

**Comment by laur89** [comment]
Not what was asked, but `F6` also toggles the address bar focus, but in that case using the escape hatch is arguably comfier.
